test/smoke/packages/cert-manager/1.5.3/cert-manager-test.sh

```diff
@@ -73,17 +82,18 @@ spec:
     - www.example.com
 EOF
 
-sleep 45s
+timeout=300
+while [[ $(kubectl get certificate/example-com -n ${NAMESPACE} -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" && ${timeout} -ne 0 ]]; do echo "waiting for certificate" && sleep 1 && timeout=${timeout}-1 ; done
```

Comment:
We can add some new lines here to avoid a very long single line. And if possible capture the `kubectl get` command in a variable or function and use it to keep it shorter and easier to understand what's going on

Also, we can have a bit more of `sleep` time to avoid checking every 1 second and printing to wait, as that will be a lot of printing but yeah, it will be a quick check

On a side note, I was thinking if this whole while loop along with body can be replaced with a `kubectl wait`, I think it could be and it's more easier to read and reason out `kubectl wait` and lesser code to maintain. So if that's possible, I would prefer it any day

Reply:
I haven’t tried `kubectl wait` with custom resources like `Certificate` but reading the `kubectl wait -h` output, it just says it works with resources in resource. You can dig deep. I’ll also check it out sometime